src/components/internals/calendar-body/calendar-body.js

```diff
@@ -12,22 +12,18 @@ import {
 
 export const ClearSection = props => (
   <div
-    onClick={props.isDisabled ? undefined : props.onClear}
+    onClick={props.onClear}
     css={getClearSectionStyles(props)}
     aria-label="clear"
   >
-    {!props.isDisabled && (
```

Comment:
If we're already hiding the icon when `isDisabled`, then there's no point rendering the `ClearSection` at all. Which is why I removed all instances of the prop from here.